### ISSUE TITLE
1dcomplex-schrodinger does not converge

### ISSUE THREAD

**Issue by EyalRozenberg1** [pierremtb/PINNs-TF2.0#1]
Hay
the 1dcomplex-schrodinger does not converge to results presented in the article.
maybe there is additional commit missing?

ps. I did sanity chack by running 1d-burgers from current commit and it actually worked well in the same environment.

thanks

**Comment by pierremtb** [comment]
Hey, sorry I haven't maintained this in a while, but as far as I remember, I didn't get this one to converge indeed

**Comment by levimcclenny** [comment]
I just re-built a version of this and had to make some changes to the network structure to achieve convergence. Your layer input regularization (for which you use a lambda layer in Keras) isn't applied to every layer, which is likely causing an issue with the size of the gradients. Since the tanh function is used (which is really only good in a small range), a large value in the output of the dense layer is likely causing the gradients to vanish as the training progresses. I had to create a custom layer in Keras to implement the regularization Raissi used, which fixed the issue for me. However, I did this for continuous inference of Allan-Cahn, not schrodingers. So you may achieve better convergence in this version if you change your activations from tanh to LeakyReLU. Or build a custom class layer instead of using the built-in Dense layer from Keras (which is what I did) to fix this issue.

**Comment by vm-pierre** [comment]
Thanks for the input @levimcclenny!
